bundles/binding/org.openhab.binding.modbus/src/main/java/org/openhab/binding/modbus/internal/ModbusGenericBindingProvider.java

```diff
@@ -152,14 +153,19 @@ State getItemState() {
             return item.getState();
         }
 
+        public void setItemState(State state) {
+            ((GenericItem) item).setState(state);
+            ;
```

Comment:
Could the binding config just hold onto a `private State currentState` member, and do something like:
```
            if (slave.isUpdateUnchangedItems() || !newState.equals(config.getCurrentState())) {
                eventPublisher.postUpdate(itemName, newState);
                config.setCurrentState(newState);
            }
```